src/orquestra/sdk/_base/cli/_dorq/_login/_login.py

```diff
@@ -37,26 +42,67 @@ def __init__(
         self._config_repo: _repos.ConfigRepo = config_repo
         self._runtime_repo: _repos.RuntimeRepo = runtime_repo
         self._login_server: LoginServer = login_server
+        self._config_resolver: _arg_resolvers.ConfigResolver = config_resolver
 
-    def on_cmd_call(self, url: str, token: t.Optional[str], ce: bool):
+    def on_cmd_call(
+        self,
+        config: t.Optional[str],
+        url: t.Optional[str],
+        token: t.Optional[str],
+        ce: bool,
+    ):
+        """
+        Call the login command action, catching any exceptions that arise.
+        """
         try:
-            self._on_cmd_call_with_exceptions(url, token, ce)
+            self._on_cmd_call_with_exceptions(config, url, token, ce)
         except Exception as e:
             self._exception_presenter.show_error(e)
 
     def _on_cmd_call_with_exceptions(
         self,
-        url: str,
+        config: t.Optional[str],
+        url: t.Optional[str],
         token: t.Optional[str],
         ce: bool,
     ):
         """
         Implementation of the command action. Doesn't catch exceptions.
         """
+        assert bool(config) ^ bool(url), (
+            "orq login action was called with arguments "
+            f"'config = {config}, url = {url}'. "
+            "Exactly one of these arguments must be provided, but this constraint "
+            "should have been handled at CLI entry. Please report this as a bug."
+        )
+
+        _url: str
+        if config:
+            loaded_config = self._config_repo.read_config(
```

Comment:
Question: What happens when there is no config stored?

Reply:
Depending on exactly what you mean here:
- When the named config is not stored in the file:
```
> orq login -c foo  
[?] No config 'foo' found in file. Please select a valid config: prod-d
 > prod-d
   127
   moreau
```
- when there are no suitable configs stored in the file:
```
$ orq login -c foo
Traceback (most recent call last):
  File "/Users/benjaminmummery/Documents/Projects/orquestra-workflow-sdk/src/orquestra/sdk/_base/cli/_dorq/_login/_login.py", line 59, in on_cmd_call
    self._on_cmd_call_with_exceptions(config, url, token, ce)
  File "/Users/benjaminmummery/Documents/Projects/orquestra-workflow-sdk/src/orquestra/sdk/_base/cli/_dorq/_login/_login.py", line 83, in _on_cmd_call_with_exceptions
    self._config_resolver.resolve_stored_config_for_login(config)
  File "/Users/benjaminmummery/Documents/Projects/orquestra-workflow-sdk/src/orquestra/sdk/_base/cli/_dorq/_arg_resolvers.py", line 88, in resolve_stored_config_for_login
    return self._prompter.choice(valid_configs, message=message)
  File "/Users/benjaminmummery/Documents/Projects/orquestra-workflow-sdk/src/orquestra/sdk/_base/cli/_dorq/_ui/_prompts.py", line 82, in choice
    raise exceptions.NoOptionsAvailableError(message)
orquestra.sdk.exceptions.NoOptionsAvailableError: No config 'foo' found in file. Please select a valid config

No config 'foo' found in file. Please select a valid config:
No options are available.
```